pkg/datasources/external_functions.go

```diff
@@ -56,42 +57,59 @@ var externalFunctionsSchema = map[string]*schema.Schema{
 
 func ExternalFunctions() *schema.Resource {
 	return &schema.Resource{
-		Read:   ReadExternalFunctions,
-		Schema: externalFunctionsSchema,
+		ReadContext: ReadContextExternalFunctions,
+		Schema:      externalFunctionsSchema,
 	}
 }
 
-func ReadExternalFunctions(d *schema.ResourceData, meta interface{}) error {
+func ReadContextExternalFunctions(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
 	databaseName := d.Get("database").(string)
 	schemaName := d.Get("schema").(string)
+	req := sdk.NewShowExternalFunctionRequest()
 
-	currentExternalFunctions, err := snowflake.ListExternalFunctions(databaseName, schemaName, db)
-	if errors.Is(err, sql.ErrNoRows) {
-		// If not found, mark resource to be removed from state file during apply or refresh
-		log.Printf("[DEBUG] external functions in schema (%s) not found", d.Id())
-		d.SetId("")
-		return nil
-	} else if err != nil {
-		log.Printf("[DEBUG] unable to parse external functions in schema (%s)", d.Id())
+	externalFunctions, err := client.ExternalFunctions.Show(ctx, req)
```

Comment:
in old implementation there was showing in schema: ```stmt := fmt.Sprintf(`SHOW EXTERNAL FUNCTIONS IN SCHEMA "%s"."%v"`, databaseName, schemaName)```, so I guess we should stick with this (ofc after making them optional it has to be a bit more sophisticated, but still, we should use them).

Reply:
make schemaName required with databaseName

Reply:
According to documentation, this is not supported https://docs.snowflake.com/en/sql-reference/sql/show-external-functions.